Decisions/scs-xxxx-v1-ssd-flavors.md

```diff
@@ -0,0 +1,252 @@
+(ceph in the SCS reference implementation) an unneeded and undesired property.
+Furthermore, write access to networked and replicated storage typically incurs
+a certain latency, as the writes can only be acknowledged once all the replicas
+have confirmed that the data has hit stable storage. Write latency is critical
+for e.g. relational database performance.
+
+The main purpose for the IaaS layer in SCS is to perform as a solid foundation
+to provide and manage kubernetes container clusters in a multi-tenant scenario.
+As such the standards at the IaaS layer should ensure that all the needed
+types of resources are available for the container clusters. This is not
+currently the case: In a scenario with multiple k8s control-plane nodes set
+up via kubeadm (as part of the k8s cluster-api automation), the control plane
+nodes each run an etcd instance and together form an etcd cluster.
+
+etcd is sensitive to scheduling, network and storage latencies. While network
+latencies and scheduling latencies have not been observed to be an issue in
+clusters within one cloud region, the storage latency is. With remote networked
+storage as delivered from ceph, the long tail of write latency causes etcd
+to often time out heartbeats, causing a new leader election with a leader
+change, preventing control plane changes on k8s for a few seconds. Too many
+leader changes can slow down cluster operation and even bring it to a halt.
+
+The etcd requirements [are well documented](https://etcd.io/docs/v3.5/op-guide/hardware/#example-hardware-configurations).
+In particular, over a hundred of *sequential* IOPS are recommended. This
+requires write latencies in the range of a single-digit ms (or better).
+
+# Design Considerations
+
+## Options considered
+
+### One-node etcd (backed by redundant storage)
+
+If k8s uses only one control plane node, there will only be only one etcd node,
+avoiding timed out heartbeats. Single node control planes are typically not
+recommended for production workloads though. They are limited with respect
+to control plane performance, have a higher chance to fail (as a single node failure
+can create cluster control-plane downtime) and can not undergo rolling upgrades.
+
+Though not the normal setup with kubeadm, it is possible to use a multi-node
+control plane using a single-node etcd. This shares some of the challenges of
+single-node control-planes, although recovery may be faster to perform at least
+in scenarios where the etcd backend storage is redundant and not affected by the
+single-node outage.
+
+Neither scenario fulfills typical requirements for production workloads.
+
+### RAM (tmpfs) etcd storage
+
+etcd could keep its database in volatile memory (e.g. on a tmpfs filesystem).
+For multi-node etcd clusters, this would actually work, as long as at least one
+cluster member stays alive. A loss of power affecting all nodes or a hardware
```

Comment:
I remember that in a team call, this was a topic of discussion.

Regarding the `this would actually work, as long as at least one cluster member stays alive` bit:

According to my own experiences, and most importantly, answers of contributors in the etcd issues (https://github.com/etcd-io/etcd/issues/7445), this is not really all that easy.

According to this information, this would not work unless shut-down etcd members are manually removed, which I do not think is the default.

Reply:
Alright, I improved the wording, mentioning that this could be made work and that proper care needs to be taken to remove shut-down nodes.